src/core/loader/ncch.cpp

```diff
@@ -246,20 +243,12 @@ ResultStatus AppLoader_NCCH::LoadExeFS() {
     if (!file.IsOpen())
         return ResultStatus::Error;
 
-    // Reset read pointer in case this file has been read before.
-    file.Seek(0, SEEK_SET);
+    // Reset read pointer to ncch_offset case this file has been read before.
+    file.Seek(ncch_offset, SEEK_SET);
```

Comment:
I asked where it is set and you set that it's set to 0 by ncch.h - no mention of the subclass..